openedx/frontend-app-authoring
# feat: add library component picker

## Description

This PR implements the components

### Select Library
![image](https://github.com/user-attachments/assets/8493b272-9f02-47a1-8c63-3bc276041946)

### Pick Component
![image](https://github.com/user-attachments/assets/55581dcd-d343-4e61-8ed8-3ee43874d19f)

## Additional Information
Part of:
- https://github.com/openedx/frontend-app-authoring/issues/1337

Depends on:
- https://github.com/openedx/edx-platform/pull/35636

## Testing instruction
- Access  http://apps.local.edly.io:2001/course-authoring/component-picker?parentLocator=block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical1
- `block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical1` should be the usageKey of a `Unit`
- Select a Library
- Navigate through the library and enter some collections
- The library should be in read-only mode, not allowing any change in the components/collections
- Click on the "+ Add" button (or in the "Add to Course") and check if the component was added to the `Unit` provided

___
Private-ref: [FAL-3876](https://tasks.opencraft.com/browse/FAL-3876)
